app/src/server/api/routers/blackmarket.ts

```diff
@@ -527,3 +622,82 @@ export const fetchActiveUserOffers = async (client: DrizzleClient, userId: strin
     where: and(eq(ryoTrade.creatorUserId, userId), isNull(ryoTrade.purchaserUserId)),
   });
 };
+
+// Helper function to get rolled elements from actionLog
+async function getRolledElements(client: DrizzleClient, userId: string, elementType?: "primary" | "secondary"): Promise<{ primary: string[], secondary: string[] }> {
```

Comment:
We try never to use explicit return types for maintainability. For consistency, use arrow-functions, i.e. `const getRolledElements = ...`